### PR TITLE
add countby_many

### PR DESCRIPTION
**Pull request by davidstrouk**

### CODE REVIEW

File: gamla/functional.py [hunk 1]
```diff
@@ -366,3 +366,36 @@ def eq_by(f, value_1, value_2):
...
+     'd': 1,
+     'n': 1,
+     'h': 1,
+     'f': 1,
+     'k': 1}
+    """
+    return toolz.pipe(
+        it,
+        curried.mapcat(f),
+        curried.groupby(toolz.identity),
```
Comment (uriva) [inline gamla/functional.py:399]: consider reduce_by_many for memory reasons

File: gamla/functional.py [hunk 2]
```diff
@@ -366,3 +366,36 @@ def eq_by(f, value_1, value_2):
 
 
 eq_str_ignore_case = eq_by(str.lower)
+
+
+@toolz.curry
```
Comment (uriva) [inline gamla/functional.py:371]: consider gamla.curry for performance reasons

File: gamla/functional.py [hunk 3]
```diff
@@ -366,3 +366,36 @@ def eq_by(f, value_1, value_2):
...
+def countby_many(f, it):
```
Comment (uriva) [inline gamla/functional.py:372]: consider adding a test

**Review by uriva — APPROVED** [review]

File: gamla/functional.py (hunk 1 above)
Comment (davidstrouk) [reply gamla/functional.py:399]: do you mean `groupby_many`? there is no reduce_by_many in gamla

Comment (davidstrouk) [reply gamla/functional.py:399]: * `groupby_many_reduce` ?

Comment (davidstrouk) [reply gamla/functional.py:399]: used `groupby_many_reduce`

**Pull request merged**